tests/e2e/driver.go

```diff
@@ -0,0 +1,245 @@
+	storageframework.TestDriver
+	GetRestoreSnapshotStorageClass(ctx context.Context) *storagev1.StorageClass
+}
+
+type fsxDriver struct {
+	driverInfo                storageframework.DriverInfo
+	parameters                map[string]string
+	restoreSnapshotParameters map[string]string
+	snapshotParameters        map[string]string
+	createInput               any
+	deleteInput               any
+}
+
+var _ FSxTestDriver = &fsxDriver{}
+var _ storageframework.TestDriver = &fsxDriver{}
+var _ storageframework.PreprovisionedVolumeTestDriver = &fsxDriver{}
+var _ storageframework.PreprovisionedPVTestDriver = &fsxDriver{}
+var _ storageframework.DynamicPVTestDriver = &fsxDriver{}
+var _ storageframework.CustomTimeoutsTestDriver = &fsxDriver{}
+
+type FSxResource struct {
+	deleteInput any
+}
+
+type FSxFilesystem struct {
+	fsxResource  FSxResource
+	filesystemId string
+	dnsName      string
+	RootVolumeId string
+}
+
+type FSxVolume struct {
+	fsxResource FSxResource
+	volumeId    string
+	dnsName     string
+	volumePath  string
+}
+
+func InitFSxCSIDriver(parameters map[string]string, restoreSnapshotParameters map[string]string, snapshotParameters map[string]string, createInput any, deleteInput any) storageframework.TestDriver {
+	var min string
+	var max string
+	if parameters[RESOURCETYPE] == RESOURCETYPE_FILESYSTEM {
+		min = "64Gi"
+		max = "512Ti"
+	} else if parameters[RESOURCETYPE] == RESOURCETYPE_VOLUME {
+		min = "1Gi"
+		max = "1Gi"
+	}
+
+	return &fsxDriver{
+		driverInfo: storageframework.DriverInfo{
+			Name:            driver.DriverName,
+			SupportedFsType: sets.NewString(""),
+			SupportedSizeRange: e2evolume.SizeRange{
+				Min: min,
+				Max: max,
+			},
+			Capabilities: map[storageframework.Capability]bool{
+				storageframework.CapPersistence: true,
+				storageframework.CapExec:        true,
+				storageframework.CapMultiPODs:   true,
+				storageframework.CapRWX:         true,
+				//storageframework.CapSnapshotDataSource:  true, //Base TestSuites doesn't allow for different storage class parameters during restore, therefore skipping in favor of custom suite
+				storageframework.CapControllerExpansion: true,
+			},
+		},
+		parameters:                parameters,
+		restoreSnapshotParameters: restoreSnapshotParameters,
+		snapshotParameters:        snapshotParameters,
+		createInput:               createInput,
+		deleteInput:               deleteInput,
+	}
+}
+
+func (d *fsxDriver) GetDriverInfo() *storageframework.DriverInfo {
+	return &d.driverInfo
+}
+
+func (d *fsxDriver) SkipUnsupportedTest(pattern storageframework.TestPattern) {
+	if d.parameters[RESOURCETYPE] == RESOURCETYPE_VOLUME {
+		if pattern.AllowExpansion != false {
+			skipper.Skipf("FSxZ Volumes does not support volume expansion -- skipping")
+		}
+	}
+}
+
+func (d *fsxDriver) PrepareTest(ctx context.Context, f *framework.Framework) *storageframework.PerTestConfig {
+	return &storageframework.PerTestConfig{
+		Driver:    d,
+		Prefix:    "fsx",
```

Comment:
How does the testing framework utilize this Prefix? I see it's not referenced elsewhere in this PR. Should this be fsx-openzfs?

Reply:
Yea, pretty much K8s resources that are dynamically created has this prefixed in the name. I can change it to `fsx-openzfs`, it's mostly for debugging purposes.